Chap_API_Server.tex

```diff
@@ -160,15 +160,12 @@ \subsection{\code{PMIx_server_register_nspace}}
 If more than one application is included in the namespace, then the host environment is also \textit{required} to provide the following attributes:
 
 \begin{itemize}
-    \item for each application:
```

Comment:
As mentioned in the PR header there is no such entity in PMIx as an application. 
For example PRRTE is storing this data on per-process basis:
https://github.com/pmix/prrte/blob/master/orte/orted/pmix/pmix_server_register_fns.c#L418:L448
This separation seems misleading for me.

Reply:
Oh my - I really disagree. There most certainly is the concept of an application in PMIx. I agree that PRRTE may be storing things differently than described here, but that may be an issue with PRRTE or could be an error here. I'd have to look more carefully.

Remember, we are doing this after-the-fact. We already had to make adjustments to the PMIx library to comply with the standard as it evolved - not reason to believe that similar changes to PRRTE and/or the Slurm plugin will be required.

Reply:
> Oh my - I really disagree. There most certainly is the concept of an application in PMIx.
There is a mention of the application in PMIx, but no program object in API that corresponds to it. At least the way I read the standard.
For example, how can client request information about the size of applicatoin number 2?

It can reference jobs (by namespaces) so can request the size of job corresponding to namespace "ns2" by calling
```C
PMIx_Get("ns2", PMIX_RANK_WILDCARD, PMIX_JOB_SIZE, ...);
```
It can ask for some process-specific information, for example for process 2 it can call
```
PMIx_Get("ns1", 2, PMIX_NODEID, ...);
```
But how will it reference specific application number?

Reply:
You use the info keys - I can add an example to the wording. We use it in various places today.

Reply:
Then I guess application-level information should be grouped the same way proc-level keys are being grouped today.
But I don't see PMIX_APP_DATA key for that.

Reply:
I mean grouped by host RM when submitting it

Reply:
I'll try to clarify the text and provide an example or two.